src/qtmaximise.h

```diff
@@ -0,0 +1,35 @@
+// Copyright 2015, Christopher J. Foster and the other displaz contributors.
+// Use of this code is governed by the BSD-style license found in LICENSE.txt
+
+#ifndef QT_MAXIMISE_H_INCLUDED
+#define QT_MAXIMISE_H_INCLUDED
+
+#include <QAction>
+#include <QMainWindow>
+
+class QActionMaximise : public QAction
+{
+    Q_OBJECT
+
+    public:
+        QActionMaximise(QMainWindow* mainWindow = 0) : QAction(tr("Maximise"),mainWindow), m_mainWindow(mainWindow)
+        {
+            setShortcut(Qt::Key_F11);
+            connect(this, SIGNAL(triggered(bool)), this, SLOT(handleTriggered(bool)));
+        }
+
+    public slots:
+        void handleTriggered(bool checked)
+        {
+            const bool maximised = m_mainWindow->windowState()&Qt::WindowMaximized;
+            if (maximised)
+                m_mainWindow->showNormal();
+            else
+                m_mainWindow->showMaximized();
+        }
```

Comment:
Ah, but Qt is object-oriented.  And QActionMaximize is legitimately specific to QMainWindow.
What bothers me more is using Australian english for a Maximise class calling Maximize.